packages/lit-dev-content/site/tutorials/content/svg-templates/04.md

```diff
@@ -0,0 +1,96 @@
+This section of the tutorial covers the basics of the `<pattern>`
+element in SVG including how to:
+
+- Create a `<pattern>` 
+- Apply a `<pattern>` as a fill
+
+### Learn
+
+In SVG, the `<pattern>` element visually repeats children across
+a 2d plane.
+
+The `patternUnits` property defines the relative coordinate geometry
+used to paint a pattern. Setting `patternUnits` to `userSpaceOnUse` makes
+geometry relative to the user space. In this case, user space is the
+DOM.
+
+```ts
+const helloPattern = svg`
+	<pattern patternUnits="userSpaceOnUse">
+		${createTile()}
+	</pattern>
+`;
+```
+
+The `<pattern>` element can be given an `id` and referenced as a fill
+in other SVG elements as shown in the example below:
+
+```ts
+const helloPattern = svg`
+	<pattern
+		id="hello-pattern"
+		patternUnits="userSpaceOnUse">
+		${createTile()}
+	</pattern>
+`;
+
+const helloPatternFill - svg`
+    <rect fill="url(#hello-pattern)" width="200" height="200"></rect>
+`;
+```
+
+### Apply
+
+Create a function called `createRepeatpattern` that returns a pattern
+element. Call `createTile` inside a `<pattern>` element. Give the
+`<pattern>` an `id` of `pattern-rounds`.
+
+```ts
+const createRepeatPattern = () => svg`
+	<pattern id="pattern-rounds">
+		${createTile()}
+	</pattern>
+`;
+```
+
+Provide [pattern attributes](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/pattern#attributes)
+like `height` and `width` to define how a tiles are painted. The example
+below creates a 200px square tile with a vertical and horizontal offset of
+`-10`. The pattern is painted using geometry defined by `userSpaceOnUse`.
+
+```ts
+const createRepeatPattern = () => svg`
+	<pattern
+		id="pattern-rounds"
+		x="-10"
+		y="-10"
+		width="200"
+		height="200"
+		patternUnits="userSpaceOnUse">
+		${createTile()}
+	</pattern>
+`;
+```
+
+Add two `<rect>` elements to the `render` method in  `repeat-pattern`. One
+rect will be a solid white background. The other rect will have a `fill`
+referencing the `<pattern>` element by `#pattern-rounds`.
+
+```ts
```

Comment:
you don't say to put `${createRepeatPattern()}` in the defs in render

Reply:
i let the audience know! sg